jet/cdc/src/main/java/com/hazelcast/samples/jet/cdc/CdcRealTimeAnalysisDemo.java

```diff
@@ -0,0 +1,144 @@
+ * in the upstream system and will update some analytics based on actions done by customers.
+ *
+ * To run this example you have to have PostgreSQL database set up. You can quickly do this, by running in your terminal:
+ * <pre>
+ *     docker run -d --rm --name cdc-postgres -p 5432:5432 \
+ *      -e POSTGRES_DB=postgres -e POSTGRES_USER=postgres \
+ *      -e POSTGRES_PASSWORD=cdcpwd1337 debezium/example-postgres:1.9.3.Final
+ * </pre>
+ *
+ * The image comes with some data already filled. Later you can add more data manually to test the queries.
+ *
+ * Results of the pipeline are visible in the {@code CustomerStatsReport} IMap or via {@code CustomerStatsReport} SQL
+ * mapping.
+ */
+public class CdcRealTimeAnalysisDemo {
+
+    @SuppressWarnings({ "resource", "checkstyle:MethodLength" })
+    public static void main(String[] args) {
+        var config = new Config();
+        config.getJetConfig().setEnabled(true);
+        config.getSerializationConfig()
+                .getCompactSerializationConfig()
+                .register(CustomerStatsReport.class)
+                .setEnabled(true);
+
+        HazelcastInstance hz = Hazelcast.newHazelcastInstance(config);
+        JetService jet = hz.getJet();
+        // trigger initialization
+        hz.getMap("CustomerStatsReport");
+
+        hz.getSql().execute(
+          """
+                  create mapping CustomerStatsReport (
+                    customerId int,
+                    customerFirstName varchar,
+                    customerLastName varchar,
+                    ordersTotal int,
+                    itemsTotal int,
+                    itemsAvg double)
+                  type IMap options (
+                    'keyFormat' = 'int',
+                    'valueFormat' = 'compact',
+                    'valueCompactTypeName'='CustomerStatsReport'
+                  );
+                  """
+        );
+
+        Pipeline pipeline = Pipeline.create().readFrom(
+                        postgres("inventory-db")
+                                .setDatabaseName("postgres")
+                                .setSchemaWhitelist("inventory")
+                                .setDatabaseAddress("localhost")
+                                .setDatabaseUser("postgres")
+                                .setDatabasePassword("cdcpwd1337")
+                                .setDatabasePort(5432)
+                                .setTableWhitelist("inventory.customers", "inventory.orders")
+                                .build())
+                .withoutTimestamps()
+                .flatMap(record -> {
+                    if (isOrder(record) && record.operation() == Operation.UPDATE) {
+                        return traverseItems(
+                                orderEvent(record.newValue().toObject(Order.class), INSERT),
+                                orderEvent(record.oldValue().toObject(Order.class), DELETE)
```

Comment:
Shouldn't you traverse this in reversed order? Otherwise, you might temporarily get larger result with both old and new version of the order.

Reply:
Yeah, that's a good idea. In general the timing shouldn't really be an issue, but worth changing, just in case :)